common/app/views/fragments/image.scala.html

```diff
@@ -10,6 +10,19 @@
     isImmersiveMainMedia: Boolean = false
 )
 
+@immersivePortraitSource(largestImage: model.ImageAsset, hidpi: Boolean = false) = {
+    <source media="@if(hidpi) {
+        (orientation: portrait) and (-webkit-min-device-pixel-ratio: 1.25), (orientation: portrait) and (min-resolution: 120dpi)
+    } else {
+        (orientation: portrait)
+    }"
+    sizes="@{100 * (largestImage.width.toDouble / largestImage.height.toDouble)}vh"
+    srcset="@{(500 to 4000 by 250)
```

Comment:
:smiley: But let’s this last step actually be the image’s original width, so that the situation I was afraid of won’t occur (last step being 248px smaller that it could just because image’s widths is `last step-1px`). If that makes sense...